pkg/liqoctl/install/kubeadmprovider/provider.go

```diff
@@ -0,0 +1,85 @@
+package kubeadmprovider
+
+import (
+	"context"
+	"fmt"
+
+	flag "github.com/spf13/pflag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+// NewKubeadmProviderConstructor initializes a new KubeadmProvider struct.
+func NewKubeadmProviderConstructor() *KubeadmProvider {
```

Comment:
```suggestion
func NewKubeadmProviderConstructor() ProviderInterface {
```

better to return the interface instead of the specific implementation, since it is simply accessible from other packages (it is exported) and we don't need implementation details

my only doubt is that it can cause some kind of circular import, it may be useful to define the interface in a separate package